plugin-tester-scala/src/main/scala/example/myapp/helloworld/AuthenticatedGreeterServer.scala

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (C) 2018-2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+//#full-server
+package example.myapp.helloworld
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.UseHttp2.Always
+import akka.http.scaladsl.model.{ HttpRequest, HttpResponse }
+import akka.http.scaladsl.server.{ Directive0, Route, RouteResult }
+import akka.http.scaladsl.server.Directives._
+import akka.http.scaladsl.{ Http, Http2, HttpConnectionContext }
+import akka.stream.{ ActorMaterializer, Materializer }
+import com.typesafe.config.ConfigFactory
+import example.myapp.helloworld.grpc._
+
+import scala.concurrent.{ ExecutionContext, Future }
+
+object AuthenticatedGreeterServer {
+
+  def main(args: Array[String]): Unit = {
+    // Important: enable HTTP/2 in ActorSystem's config
+    // We do it here programmatically, but you can also set it in the application.conf
+    val conf = ConfigFactory.parseString("akka.http.server.preview.enable-http2 = on")
+      .withFallback(ConfigFactory.defaultApplication())
+    val system = ActorSystem("HelloWorld", conf)
+    new AuthenticatedGreeterServer(system).run()
+    // ActorSystem threads will keep the app alive until `system.terminate()` is called
+  }
+}
+
+class AuthenticatedGreeterServer(system: ActorSystem) {
+
+  def run(): Future[Http.ServerBinding] = {
+    // Akka boot up code
+    implicit val sys: ActorSystem = system
+    implicit val mat: Materializer = ActorMaterializer()
+    implicit val ec: ExecutionContext = sys.dispatcher
+
+    // Create service handlers
+    val handler: HttpRequest => Future[HttpResponse] =
+      GreeterServiceHandler(new GreeterServiceImpl())
+
+    // As a Route
+    val handlerRoute: Route = {
+      ctx => handler(ctx.request).map(RouteResult.Complete)
+    }
```

Comment:
Easy enough but perhaps we could provide a `handler.asDirective`? (not scope of this PR though)

Reply:
I agree it would be nice if this would be easier. The reason this is not currently there is because we currently only depend on `akka-http-core`, and directives are an `akka-http` concept - so that's why those types are not used there. Perhaps we should just pull in all of akka-http at some point, but that's outside the scope of this PR ;).

Reply:
Ah, didn't think of that, d'oh!